modules/i2s/api/i2s_tdm_slave.h

```diff
@@ -0,0 +1,224 @@
+ * \param tx_offset             The number of bclks from FSYNC transition to the MSB
+ *                              of Slot 0
+ * \param slave_bclk_pol        The polarity of bclk
+ * \param tdm_post_port_init    Callback to be called just after resource init.
+ *                              Allows for modification of port timing for >15MHz clocks.
+ *                              Set to NULL if not needed.
+ */
+void i2s_tdm_slave_tx_16_init(
+        i2s_tdm_ctx_t *ctx,
+        i2s_callback_group_t *i2s_cbg,
+        port_t p_dout,
+        port_t p_fsync,
+        port_t p_bclk,
+        xclock_t bclk,
+        uint32_t tx_offset,
+        i2s_slave_bclk_polarity_t slave_bclk_polarity,
+        tdm_post_port_init_t tdm_post_port_init);
+
+/**
+ * I2S TDM TX 16 ch slave task
+ *
+ * This task performs I2S TDM slave on the provided context which was
+ * initialized with i2s_tdm_slave_tx_16_init(). It will perform
+ * callbacks over the i2s_callback_group_t callback group to get
+ * data from the application using this component.
+ * 
+ * This thread outputs assumes 1 output port, 32b word length,
+ * 32b channel length, and 16 channels per frame.
+ *
+ * The component performs I2S TDM slave so will expect the fsync and
+ * bit clock to be driven externally.
+ *
+ * \param ctx             A pointer to the I2S TDM context to use.
+ */
+void i2s_tdm_slave_tx_16_thread(
+        i2s_tdm_ctx_t *ctx);
+
+/**@}*/ // END: addtogroup hil_i2s_tdm_slave_tx16
+
+/**
+ * \addtogroup hil_i2s_tdm_slave hil_i2s_tdm_slave
+ *
+ * The public API for using the HIL I2S TDM generic slave.
+ * @{
+ */
+
+/**
+ * I2S generic TDM slave context initialization
+ *
+ * This prepares a context for I2S TDM slave on the provided pins.
+ * 
+ * The resulting context can be used with an I2S thread call.
+ *
+ * \param ctx                   A pointer to the I2S TDM context to use.
+ * \param i2s_cbg               The I2S callback group pointing to the application's
+ *                              functions to use for initialization and getting and receiving
+ *                              frames.  For TDM the app_data variable within this
+ *                              struct is NOT used.
+ * \param p_dout                An array of data output ports.  MUST be 1b ports
+ * \param num_out               The number of output data ports
+ * \param p_din                 An array of data input ports.  MUST be 1b ports
+ * \param num_in                The number of input data ports
+ * \param p_fsync               The fsync input port. MUST be a 1b port
+ * \param p_bclk                The bit clock input port. MUST be a 1b port
+ * \param bclk                  A clock that will get configured for use with
+ *                              the bit clock
+ * \param tx_offset             The number of bclks from FSYNC transition to the MSB
+ *                              of Slot 0 
+ * \param fsync_len             The length of the FSYNC in BCLKS
```

Comment:
I don't think ticks is a useful term here since we are talking HW interfaces, so have changed to:

The number of BCLKS from FSYNC transition to the MSB of Slot 0